src/HDF5Msg.chpl

```diff
@@ -50,20 +50,12 @@ module HDF5Msg {
     proc lshdfMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
         // reqMsg: "lshdf [<json_filename>]"
         var repMsg: string;
-        var (jsonfile) = payload.splitMsgToTuple(1);
+        var msgArgs = parseMessageArgs(payload, 1);
 
         // Retrieve filename from payload
-        var filename: string;
-        try {
-            filename = jsonToPdArray(jsonfile, 1)[0];
-            if filename.isEmpty() {
-                throw new IllegalArgumentError("filename was empty");  // will be caught by catch block
-            }
-        } catch {
-            var errorMsg = "Could not decode json filenames via tempfile (%i files: %s)".format(
-                                     1, jsonfile);                                     
-            h5Logger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-            return new MsgTuple(errorMsg, MsgType.ERROR);                                    
+        var filename: string = msgArgs.getValueOf("filename");
+        if filename.isEmpty() {
+            throw new IllegalArgumentError("filename was empty");  // will be caught by catch block
```

Comment:
same question of if we should be throwing or if we should return error message tuple

Reply:
Updated